ostree_upload_server/importers/tar.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import tempfile
+import tarfile
+
+import urllib
+import urlparse
+
+from gi.repository import GLib
+
+from os import path
+from shutil import rmtree
+
+from .base import BaseImporter
+from .util import find_repo, open_repository
+
+class TarImporter(BaseImporter):
+    MIME_TYPE = 'application/x-tar'
+
+    def _import_commit(self, target_repo, src_path_obj, refs=None):
+        if not self._source_repo_path:
+            raise RuntimeError("Cannot invoke _import_commit without calling import_to_repo first")
+
+        if not refs:
+            raise RuntimeError("Cannot invoke _import_commit without valid refs")
+
+        logging.debug("Importing %s%s into %s...", self._source_repo_path,
+                      refs,
+                      target_repo.get_path().get_uri())
+
+        options = GLib.Variant('a{sv}', {
+            'refs': GLib.Variant('as', refs),
+            'inherit-transaction': GLib.Variant('b', True),
+        })
+
+        source_repo_uri = urlparse.urljoin('file:', urllib.pathname2url(self._source_repo_path))
+        target_repo.pull_with_options(source_repo_uri,
+                                      options,
+                                      None, None)
+
+        logging.info("Importing complete.")
+
+    def import_to_repo(self):
+        logging.info('Trying to use %s extractor...', self.__class__.__name__)
+
+        dest_path = tempfile.mkdtemp('', self.__class__.__name__)
```

Comment:
Please make sure temporary files are extracted in `/var/tmp` since uploads are large and `/tmp` may be a `tmpfs`. Also, please use keyword arguments here so it's clear what you're doing (e.g., `prefix=self.__class__.__name__`).

Reply:
Yup - forgot about that. Will fix.